shippers/logstash.py

```diff
@@ -57,11 +76,17 @@ def flush(self) -> None:
 
     def _send(self, logstash_url: str, events: list[dict[str, Any]], compression_level: int) -> None:
         ndjson = "\n".join(ujson.dumps(event) for event in events)
-        response = self._session.put(
-            logstash_url,
-            data=gzip.compress(ndjson.encode("utf-8"), compression_level),
-            headers={"Content-Encoding": "gzip", "Content-Type": "application/x-ndjson"},
-        )
-        if response.status_code != 200:
-            # TODO: Change with actual handling
-            raise RuntimeError(f"Errors while sending data to Logstash. Return code {response.status_code}")
+        try:
+            self._session.put(
+                logstash_url,
+                data=gzip.compress(ndjson.encode("utf-8"), compression_level),
+                headers={"Content-Encoding": "gzip", "Content-Type": "application/x-ndjson"},
+                timeout=_TIMEOUT,
+            )
+        except RequestException as e:
```

Comment:
`RequestException` is raised only after fulfilling the retry strategy, is it?

Reply:
If Logstash responds back with any of the status codes in `_STATUS_FORCE_LIST `, yes, the request will be retried and we handle the exception if the max number of retries is exceeded. Otherwise, the request is not retried and we fail fast (e.g if we get a 401 authorization denied)